commit c0765acc46b4ae57174084609490dbe0ba869f87 (refs/changes/70/1370/2)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2017-02-07T23:32:37+00:00 (2 years, 8 months ago)
    
    NAPI-216 GetNicTag should respond with InvalidParameters on a name containing invalid chars

diff --git a/lib/models/nic-tag.js b/lib/models/nic-tag.js
index 3dbce6a..7115321 100644
--- a/lib/models/nic-tag.js
+++ b/lib/models/nic-tag.js
@@ -65,7 +65,7 @@ var CREATE_SCHEMA = {
 
 var GET_SCHEMA = {
     required: {
-        name: validate.string
+        name: validate.nicTagName
     }
 };
 
@@ -379,13 +379,14 @@ function createNicTag(app, log, params, callback) {
         }
 
         var tag = new NicTag(params);
-        app.moray.putObject(BUCKET.name, tag.name, tag.raw(),
+        app.moray.putObject(BUCKET.name, tag.name, tag.raw(), { etag: null },
             function (err2) {
             if (err2) {
-                return callback(err2);
+                callback(err2);
+                return;
             }
 
-            return callback(null, tag);
+            callback(null, tag);
         });
     });
 }
@@ -581,7 +582,7 @@ function nicTagsExist(single, opts, name, tags, callback) {
         return callback(sErr);
     }
 
-    vasync.forEachParallel({
+    vasync.forEachPipeline({
         inputs: tagArr,
         func: function _getNicTag(tag, cb) {
             getNicTag(opts.app, opts.log, { name: tag }, function (err, res) {
diff --git a/test/unit/nic-tags.test.js b/test/unit/nic-tags.test.js
index 01fe903..de911a5 100644
--- a/test/unit/nic-tags.test.js
+++ b/test/unit/nic-tags.test.js
@@ -315,7 +315,30 @@ test('Create admin nic tag - with wrong MTU', function (t) {
     });
 });
 
-// // --- Delete tests
+
+// --- Get tests
+
+
+test('Get nic tag with invalid name', function (t) {
+    NAPI.getNicTag('__ad**$@$sfasdf', function (err, obj) {
+        t.ok(err, 'error returned');
+        t.deepEqual(obj, null, 'no value returned');
+        if (!err) {
+            t.end();
+            return;
+        }
+
+        t.equal(err.statusCode, 422, '422 returned');
+        t.deepEqual(err.body, h.invalidParamErr({
+            errors: [ mod_err.invalidParam('name', INVALID_MSG) ]
+        }), 'Error body');
+
+        t.end();
+    });
+});
+
+
+// --- Delete tests
 
 
 test('Delete nic tag in use', function (t) {
